src/useField.js

```diff
@@ -38,27 +38,26 @@ const useField = (
   }: UseFieldConfig = {}
 ): FieldRenderProps => {
   const reactFinalForm: ?FormApi = React.useContext(ReactFinalFormContext)
-  // istanbul ignore next
-  if (process.env.NODE_ENV !== 'production' && !reactFinalForm) {
-    console.error(
+  if (!reactFinalForm) {
+    throw new Error(
       'Warning: useField must be used inside of a ReactFinalForm component'
     )
   }
 
   // keep ref to most recent copy of validate function
   const validateRef = React.useRef(validate)
-  validateRef.current = validate
+  React.useEffect(() => {
+    validateRef.current = validate
+  }, [validate])
```

Comment:
the effect itself is a simple assignment, so it really shouldnt be costful, I would skip the dep (`[validate]`) altogether here, if `validate` stays the same then it will also stay the same on the ref, even after assignment

Reply:
Yeah, I flip-flopped on that when writing it. I'll remove it.